cli/pcluster/commands.py

```diff
@@ -48,21 +46,24 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def _create_bucket_with_batch_resources(stack_name, resources_dir, region):
-    random_string = "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(16))
-    # bucket names must be at least 3 and no more than 63 characters long
-    s3_bucket_name = "-".join([stack_name.lower()[: 63 - len(random_string) - 1], random_string])
+def _create_bucket_with_batch_resources(s3_bucket_name, resources_dir, region):
+    """
+    Create a bucket with the given name and then upload batch resources.
 
+    Returns true if both creation and upload succeed.
+    """
     try:
         utils.create_s3_bucket(bucket_name=s3_bucket_name, region=region)
-        utils.upload_resources_artifacts(bucket_name=s3_bucket_name, root=resources_dir)
-    except boto3.client("s3").exceptions.BucketAlreadyExists:
-        LOGGER.error("Bucket %s already exists. Please retry cluster creation.", s3_bucket_name)
+    except ClientError:
+        LOGGER.critical("Unable to create internal bucket %s. Please retry cluster creation.", s3_bucket_name)
         raise
+
+    try:
+        utils.upload_resources_artifacts(bucket_name=s3_bucket_name, root=resources_dir)
     except Exception:
         utils.delete_s3_bucket(bucket_name=s3_bucket_name)
```

Comment:
Why is the bucket still deleted here? 
Probably the problem here was to make the function return True to allow bucket deletion outside even if the upload of resources failed. An idea could be to catch the exceptions locally and just return a tuple Boolean, Boolean with bucket creation result and bucket upload result. Or also trap the error from delete_s3_bucket.